gocourse05/hw/main.go

```diff
@@ -0,0 +1,176 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+)
+
+type Key struct {
+	x int
+	y int
+}
+
+func printField(init bool, field map[Key]string, size int) {
+	if init {
+		fmt.Println("Initing field")
+	}
+	for i := 1; i <= size; i++ {
+		s := strconv.Itoa(i)
+		if !init {
+			fmt.Print(s)
+		}
+		for j := 1; j <= size; j++ {
+			if init {
+				field[Key{x: i, y: j}] = "-"
+				s += field[Key{x: i, y: j}]
+			} else {
+				fmt.Print(field[Key{x: i, y: j}])
+			}
+		}
+		if init {
+			fmt.Println(s)
+		} else {
+			fmt.Println()
+		}
+	}
+	fmt.Print(" ")
+	for i := 1; i <= size; i++ {
+		fmt.Print(i)
+	}
+	fmt.Println()
+}
+
+func moveValidation(move string, fieldSize int) (int, int) {
+	var s []string
+	var x int
+	var y int
+	var xerr error
+	var yerr error
+	intro := "Enter move ('%s') and coordinates from 1 to %d splitting with space. Example: 1 1\n"
+	scanner := bufio.NewScanner(os.Stdin)
+LoopStart:
+	for {
+		fmt.Printf(intro, move, fieldSize)
+		fmt.Println()
+		scanner.Scan()
+		line := scanner.Text()
+		s = append(s, line)
+		if err := scanner.Err(); err != nil {
+			fmt.Fprintln(os.Stderr, "reading standard input:", err)
+		}
+		result := strings.Split(s[0], " ")
+		if len(result) < 2 {
+			break LoopStart
+		} else {
+			x, xerr = strconv.Atoi(result[0])
+			y, yerr = strconv.Atoi(result[1])
+		}
+		if xerr != nil || yerr != nil || move != move || x < 1 || y < 1 || x > fieldSize || y > fieldSize {
+			fmt.Printf(intro, fieldSize)
+		} else {
+			break
+		}
+	}
+	fmt.Println()
+	return x, y
+}
+
+func updateField(move string, answer func(string, int) (int, int), field map[Key]string, size int) bool {
+	x, y := answer(move, size)
+	currentKey := field[Key{x: x, y: y}]
+	if currentKey == "-" {
+		field[Key{x: x, y: y}] = move
+	} else {
+		fmt.Println("Key", field[Key{x: x, y: y}], "already exists!")
+		return false
+	}
+	return true
+}
+
+func checkResult(move string, field map[Key]string, size int) bool {
+	fmt.Println()
+	var h_line int
```

Comment:
it would be more readable and less error-prone if we were to declare variables next to where they're used. Currently nothing is stopping you from accidentally switching h_line with v_line in one of the checks

Reply:
I'd move the declaration of this one and v_line into their respective loops to remove the possibility of them clashing whatsoever. Unfortunately that won't work for diagonal loops, but it's still something